src/order/antichain.lean

```diff
@@ -0,0 +1,99 @@
+/-
+Copyright (c) 2020 Bhavik Mehta, Alena Gusakov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bhavik Mehta, Alena Gusakov
+-/
+import order.basic
+import data.finset
+import data.multiset.finset_ops
+
+/-!
+# Antichains
+Investigating the structure of finsets in a partial order.
+We define antichains.
+
+## Main definitions
+* `antichain` is a finset of elements in a partial order where
+  no element is less than another.
+
+* The `join` of two antichains is the set obtained by taking
+  their union and removing all elements that are less than
+  another element.
+
+* The `meet` of two antichains is the set obtained by taking
+  their intersection and removing all elements that are less than
+  another element.
+-/
+
+open partial_order
+
+universe u
+
+open_locale classical
+noncomputable theory
+
+variable {α : Type u}
+
+section
+variables [partial_order α]
+
+/--
+A set of elements of a partial order forms an antichain if no two elements
+`A` and `B` are ordered `A < B`.
+-/
+def antichain (𝒜 : finset α) : Prop := ∀ a ∈ 𝒜, ∀ b ∈ 𝒜, a ≤ b → a = b
```

Comment:
An alternative definition might be
```suggestion
def antichain : Type* := { A : finset α // ∀ a ∈ A, ∀ b ∈ A, a ≤ b → a = b }
```
which would allow you to bundle `meet` and `antichain_meet` into a single object. I don't know if this is a good idea or not.

Reply:
I thought that would make it easier but I was talked out of it, I'll try it and see how it goes

Reply:
Can I maybe turn this part into a separate PR? Trying to refactor it is proving difficult and I'd rather separate that work from the PR as it is currently